### DISCUSSION TITLE
Can't create raster zonal stats from a dataframe of quadkey/name

### DISCUSSION THREAD

**Discussion by AnthonyMockler** [thinkingmachines/geowrangler#109] [Enhancements Requests]
Hi,
I'm trying to use geowrangler.raster_zonal_stats to get zonal stats for a list of schools - The dataframe is in the format 

| index |quadkey |name | amenity (type)|
| 0 | 133332330000 | Rurubao School | school | 

**What I did:**
_raster = rzs.create_raster_zonal_stats(
    schools,
    f"data/{worldclim_2020_dset}",aggregation=dict(
        func=["mean","median"], 
        fillna=[True,True]))_

**What I expected:**
A dataframe with:

**|index | quadkey | name | amenity (type)| mean | median |**

**What I got:**

_ValueError: Can't parse quadkey as a geojson Feature object_

I can see the workaround (Use tms.xy_bounds to generate a geometry object for each row, then pass the GeoDataFrame to create_raster_zonal_stats) but this seems slow, clunky, and reduces the abstraction of the library

**Ideas:**

1) (Easiest, slowest)
Add some handling to create_raster_zonal_stats to append geometries to non-geometry-featured dataframes , e.g.

- if there's no geometry in the aoi dataframe
- Generate a geometry object for each row

2) (Faster, more elegant)
- If there's no geometry in the aoi dataframe
- Create a DF of the unique quadkeys
- Create geometry for those unique quadkeys
- Generate zonal stats for each unique quadkey
- Join the zonal stats with the aoi

**Comment by butchtm** [comment]
Hi @AnthonyMockler  thanks for feedback!

Yes, for now, the raster zonal stats feature has not been built with the use of quadkeys in mind.

So yes the current work around would be:

1) Convert the aoi (in your case, the school dataframe) so that the geometry associated with the quadkey are generated.
 2) Once  you have the geometry for the aois, then you can generate the zonal stats using the raster zonal stats module.

We will consider your suggestion as further enhancement for the raster zonal stats module.

**Reply by butchtm** [reply]
Hi @AnthonyMockler -- here's a [sample gist](https://gist.github.com/butchtm/3eff004bae150ea1b1d2dfeffdb5de17) to generate the geometry from the quadkey.
